nectar/downloaders/threaded.py

```diff
@@ -88,15 +88,16 @@ def __init__(self, config, event_listener=None, tries=DEFAULT_TRIES, session=Non
         # set of locations that produced a connection error
         self.failed_netlocs = set([])
 
-        self.session = session or build_session(config)
+        if not session:
```

Comment:
I'm not sure I understand. Do you want me to back this change out and leave a comment, or leave the fix in?